### ISSUE TITLE
Spell's "targetCondition" with defined "anyOf" doesn't work.

### ISSUE THREAD

**Issue by madaosoul** [vcmi/vcmi#2818]
I was testing my map, and randomly noticed that spell "Assume Control" from forge2k works on living creatures (it should only work on non-living and undead). The "targetCondition" of this spell looks like this:

		"targetCondition" : {
			"allOf" : {},
			"anyOf" : {
				"bonus.NON_LIVING": "normal",
				"bonus.UNDEAD": "normal"},			
			"noneOf" : {}
	},	

VCMI seemingly ignores this. The easiest way to replicate this bug is to install my mod (Heavenly Forge 2000), choose hero Everest, who starts with this spell and initiate combat with any neutral stack on the map. Assume Control will affect any type of creature (tested on 1.3.0 version and daily 1.4. version).

**Comment by rilian-la-te** [comment]
Can you share a logs?

**Comment by rilian-la-te** [comment]
@madaosoul by default spell will affect any creatures, adding "anyOf" will not offset such effect. If you want a spell to affect only undead and constructs, you need to write something like this (1.3):

```js
	"targetCondition" : {	
		"noneOf" : {
                "living" : {
                    "absolute" : false,
                    "type" : "selector",
                    "parameters" : {
                        "noneOf" : [
                            {
                                "anyOf" : [
                                    { "type" : "UNDEAD"},
                                    { "type" : "NON_LIVING}
                                ]
                            }
                        ]
                    }
                }
            }
``` 

Yes, it looks weird, I know.

**Comment by krs0** [comment]
so in fact you want to say "`notAnyOf`"?
I think you are missing explicitly this case
```
allOf
noneOf
anyOf
notAnyOf
```

**Comment by krs0** [comment]
So why not simply do not apply by default on all creatures when anyOf is used?

will be reduced to 2 clear lines:
```
                                "anyOf" : [
                                    { "type" : "UNDEAD"},
                                    { "type" : "NON_LIVING"}
                                ]
```

**Comment by rilian-la-te** [comment]
@krs0 it was not working like that at all. NotAnyOf is merged into develop, so, one AnyOf can be removed.

**Comment by madaosoul** [comment]
I don't care if it looks weird or is non-intuitive, as long as it works. And right now it doesn't. I changed the "targetCondition" according to the instructions above, and the spell can still target e.g. neutral Rogues. Here's updated .json. 
[assumeControl.zip](https://github.com/vcmi/vcmi/files/12607753/assumeControl.zip)

**Comment by rilian-la-te** [comment]
@madaosoul do you use latest beta?

**Comment by madaosoul** [comment]
No, 1.3.0. Also, it sporadically crashes the game. Almost always on daily and beta versions. I'm with @krs0 that defining "anyOf" or "allOf" should turn off the spell's default method of working on everything. And I'm 99% sure that it functioned like this at some point (on pre 1.2. versions).

**Comment by rilian-la-te** [comment]
@madaosoul on 1.1 it was not working as you described, AFAIK. Can you created a minimal mod with your changes (second iteration)? I will test it on weekend)

**Comment by madaosoul** [comment]
[onlyAssumeControl.zip](https://github.com/vcmi/vcmi/files/12608360/onlyAssumeControl.zip)

**Comment by nullkiller** [comment]
Any news here?

**Comment by rilian-la-te** [comment]
@nullkiller it needs to be tested again after #3118.
